src/chat.rs

```diff
@@ -2826,27 +2825,34 @@ pub(crate) async fn create_send_msg_job(
         msg.update_param(context).await?;
     }
 
-    ensure!(!recipients.is_empty(), "no recipients for smtp job set");
-
-    let recipients = recipients.join(" ");
+    let chunk_size = context
+        .get_configured_provider()
+        .await?
+        .and_then(|provider| provider.opt.max_smtp_rcpt_to)
+        .map_or(constants::DEFAULT_MAX_SMTP_RCPT_TO, usize::from);
+    let mut row_ids = Vec::new();
+    for recipients_chunk in recipients.chunks(chunk_size) {
+        let recipients_chunk = recipients_chunk.join(" ");
 
-    msg.subject = rendered_msg.subject.clone();
-    msg.update_subject(context).await?;
+        msg.subject = rendered_msg.subject.clone();
+        msg.update_subject(context).await?;
 
-    let row_id = context
-        .sql
-        .insert(
-            "INSERT INTO smtp (rfc724_mid, recipients, mime, msg_id)
-             VALUES           (?1,         ?2,         ?3,   ?4)",
-            (
-                &rendered_msg.rfc724_mid,
-                recipients,
-                &rendered_msg.message,
-                msg.id,
-            ),
-        )
-        .await?;
-    Ok(Some(row_id))
+        let row_id = context
+            .sql
+            .insert(
+                "INSERT INTO smtp (rfc724_mid, recipients, mime, msg_id)
+                 VALUES           (?1,         ?2,         ?3,   ?4)",
+                (
+                    &rendered_msg.rfc724_mid,
+                    recipients_chunk,
+                    &rendered_msg.message,
```

Comment:
This can blow up the `smtp` table. If this is critical, maybe we need to restructure it somehow